File: TestProjects/EUThreadManagerObjective-C/EUThreadManager/ViewController.m
//
//  ViewController.m
//  EUThreadManager
//
//  Created by Vitaliy Yarkun on 4/22/17.
//  Copyright © 2017 Vitaliy Yarkun. All rights reserved.
//

#import "ViewController.h"
#import "EUThreadManager.h"

@interface ViewController ()

@property (nonatomic, strong) EUThreadManager* threadManager;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    simpleBlock task = ^{
        for (NSInteger i = 0; i < 100000; i++) {
            NSLog(@"Current i: %li", (long)i);
        }
    };
    self.threadManager = EUThreadManager.sharedInstance;
    [self.threadManager runOnBackgroundSimpleTask:task withPriority:kDefaultPriorityExecution andResultMethod:@selector(resultMethod) fromObject:self];
}

-(void) resultMethod {
    NSLog(@"Done!");
}
@end

File: TestProjects/EUThreadManagerSwift/EUThreadManagerSwift/ViewController.swift
//
//  ViewController.swift
//  EUThreadManagerSwift
//
//  Created by Vitaliy Yarkun on 4/23/17.
//  Copyright © 2017 Vitaliy Yarkun. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    let threadManager = EUThreadManager.sharedInstance
    
    let taskToRunOnBackground = {
        
        for i in 0...10000{
            print("Current i: \(i)")
        }
    }
    
    let resultMethod = {
        
        print("Done!")
    }
    override func viewDidLoad() {
        super.viewDidLoad()
        
        threadManager.run(task: taskToRunOnBackground, with: .kDefaultPriorityExecution, and: resultMethod)
    }
}

